datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -200,6 +200,20 @@ pub fn make_current_date(
     move |_arg| Ok(ColumnarValue::Scalar(ScalarValue::Date32(days)))
 }
 
+/// Create an implementation of `current_time()` that always returns the
+/// specified current time.
+///
+/// The semantics of `current_time()` require it to return the same value
+/// wherever it appears within a single statement. This value is
+/// chosen during planning time.
+pub fn make_current_time(
+    now_ts: DateTime<Utc>,
+) -> impl Fn(&[ColumnarValue]) -> Result<ColumnarValue> {
+    let nano =
+        Duration::seconds(now_ts.num_seconds_from_midnight() as i64).num_nanoseconds();
```

Comment:
I found a bug, nanoseconds less than a second will be loss if `num_seconds_from_midnight()` is used,     I will do  this in another way

Reply:
i think you might not need Duration::seconds here.

```rust
let nano = (now_ts.num_seconds_from_midnight() as i64) * 1000000000 + (now_ts.nanosecond() as i64);
```

 [1] [nanosecond](https://docs.rs/chrono/latest/chrono/trait.Timelike.html#tymethod.nanosecond)